### PR TITLE
72 character creation modal

### PR DESCRIPTION
**Pull request by Ovorp**
**Task #72**

Created a modal where you can select your character, add a character name.

It then displays a image, stats, and passive.

Still needs styling and hooking up with the back end.

### CODE REVIEW

File: frontend/src/components/CharacterCreation/classes.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+const BARBARIAN = {
+  class: 'Barbarian',
+  str: 11,
+  dex: 8,
+  con: 15,
+  int: 6,
+  passive: 'Sustaining Rage, Random chance to heal',
+  img: 'https://i.pinimg.com/564x/f2/56/db/f256db27f25ee6d99e930c21bece150b--group-art-character-ideas.jpg',
+};
+const MAGE = {
```
Comment (bheadwhite) [inline frontend/src/components/CharacterCreation/classes.tsx:28]: would it be useful to create interfaces for each of these classes? That way you help typescript keep up with managing these.

Comment (bheadwhite) [reply frontend/src/components/CharacterCreation/classes.tsx:28]: Interface for each class and then also you can explicitly type your classList as well

File: frontend/src/components/CharacterCreation/CharacterInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import React, { ReactElement } from 'react';
+import * as CLASSES from './classes';
+// Need to look into how i'm importing.  I think doing it this way is making it harder.
+import { SKILLS } from './skills';
+
+interface Props {
+  currentClass: string;
```
Comment (bheadwhite) [inline frontend/src/components/CharacterCreation/CharacterInfo.tsx:7]: this should be `currentClass: 'Knight' | 'Rogue' | 'Barbarian' | 'Mage'` which can be an exported type from your classes file

File: frontend/src/components/CharacterCreation/CharacterInfo.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+function CharacterInfo(props: Props): ReactElement {
+  // the lines below, Current class should be an object (Record), the keys should be strings, and the values should be string or number
+
+  let currentClass: Record<string, string | number> = {
```
Comment (bheadwhite) [inline frontend/src/components/CharacterCreation/CharacterInfo.tsx:13]: if you typed the other things, this will probably clean up as well

**Review by bheadwhite — CHANGES_REQUESTED** [review]

File: frontend/src/components/CharacterCreation/classes.tsx (hunk 1 above)
Comment (Ovorp) [reply frontend/src/components/CharacterCreation/classes.tsx]: I ended up added a interface an type

File: frontend/src/components/CharacterCreation/CharacterInfo.tsx (hunk 2 above)
Comment (Ovorp) [reply frontend/src/components/CharacterCreation/CharacterInfo.tsx]: Updated so that most var have a type

**Pull request closed without merging**